java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java

```diff
@@ -663,16 +690,19 @@ public ResponseEntity<String> confirmCode(@RequestBody String code) {
 	
 	@PostMapping(path = {"/auth/change-email"})
 	@ResponseBody
-	public ResponseEntity<String> changeEmail(@RequestBody UserPasswordPost us, HttpServletRequest request) throws ServletException {
-		if (us.username != null) {
-			us.username = us.username.toLowerCase().trim();
-		}
-		if (emailSender.isEmail(us.username)) {
+	public ResponseEntity<String> changeEmail(@RequestBody Map<String, String> credentials, HttpServletRequest request) throws ServletException {
```

Comment:
Could you use more transparent way? @RequestBody String username, @RequestBody String token?